src/api/app/models/concerns/watchable.rb

```diff
@@ -0,0 +1,8 @@
+module Watchable
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :watched_items, as: :watchable, dependent: :destroy
+    has_many :users, through: :watched_items
```

Comment:
`:users` is pretty generic, can you think of a better name?

Reply:
I made it like that since i'm making it just for users right now, so that's a reference to the users table, do we have other entities at the moment that should watch "watched_projects"? if so, can you provide an example so i have a bit more context? thanks! :)

Reply:
`Project`s for instance have people *owning* them through (I'm a bit embarrassed to say) a model called  `Relationship` (yes naming is a strong-suit of us ;-). I fear that people will misunderstand this. How about `has_many :watchers, through: :watched_items, class_name: 'User'`?